boilerplate/app/screens/DemoComponentsScreen/DemoComponentsScreen.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { ReactElement, useRef, useState } from "react"
+import { FlatList, Image, Pressable, View, ViewStyle } from "react-native"
+import { DrawerLayout } from "react-native-gesture-handler"
+import { DrawerIconButton, Screen, Text } from "../../components"
+import { DemoTabScreenProps } from "../../navigators/demo-navigator"
+import { DemoItem } from "./DemoItem"
+import * as Demos from "./demos"
+
+export interface Demo {
+  name: string
+  description: string
+  useCases: ReactElement[]
+}
+
+export function DemoComponentsScreen(props: DemoTabScreenProps<"DemoComponents">) {
+  const [open, setOpen] = useState(false)
+  const drawerRef = useRef<DrawerLayout>()
+  const listRef = useRef<FlatList>()
+  const menuRef = useRef<FlatList>()
+
+  const toggleDrawer = () => {
+    if (!open) {
+      setOpen(true)
+      drawerRef.current?.openDrawer({ speed: 2 })
+    } else {
+      setOpen(false)
+      drawerRef.current?.closeDrawer({ speed: 2 })
+    }
+  }
+
+  return (
+    <DrawerLayout
+      ref={drawerRef}
+      drawerWidth={326}
+      drawerType={"slide"}
+      drawerBackgroundColor={"white"}
+      onDrawerOpen={() => setOpen(true)}
+      onDrawerClose={() => setOpen(false)}
+      renderNavigationView={() => (
+        <View style={$drawer}>
+          <View style={$logoContainer}>
+            <Image source={require("../../../assets/images/logo.png")} />
+          </View>
+          <FlatList<{ name: string; useCases: string[] }>
+            ref={menuRef}
+            data={Object.values(Demos).map((d) => ({
+              name: d.name,
+              useCases: d.useCases.map((u) => u.props.name),
+            }))}
+            keyExtractor={(item) => item.name}
+            renderItem={({ item }) => (
+              <Pressable
+                onPress={() => {
+                  listRef.current.scrollToItem({
+                    animated: true,
+                    item,
+                  })
+                  toggleDrawer()
+                }}
+                style={$menu}
+              >
+                <Text preset="menu">{item.name}</Text>
+                <View style={$subMenu}>
+                  {item.useCases.map((u) => (
+                    <View key={u} style={$subMenuItem}>
+                      <Text>{u}</Text>
+                      <Image source={require("../../../assets/images/caratRight.png")} />
```

Comment:
Minor nitpick - let's change `carat` -> `caret`, although a diamond icon would be kinda cool. :) 

Also, we have an `Icon` component you can use which includes the assets. You can move the icon there for now -  I'll move the assets around later when i work on the component.